src/manifest.rs

```diff
@@ -64,17 +64,28 @@ pub(crate) fn get_project_dir_from_manifest_path(
 }
 
 /// A [package.metadata] or [workspace.metadata] table with
-/// `cargo-semver-checks` lint entries stored in `config`
+/// `cargo-semver-checks` config entries stored in `config`
 #[derive(Debug, Clone, Deserialize)]
-pub(crate) struct LintTable {
+pub(crate) struct MetadataTable {
+    /// Holds the `cargo-semver-checks` table, if it is declared.
     #[serde(default, rename = "cargo-semver-checks")]
-    pub(crate) config: Option<BTreeMap<String, OverrideConfig>>,
+    pub(crate) config: Option<SemverChecksTable>,
 }
 
-impl LintTable {
-    pub fn into_overrides(self) -> Option<OverrideMap> {
-        self.config
-            .map(|config| config.into_iter().map(|(k, v)| (k, v.into())).collect())
+/// A [cargo-semver-checks] config table in [package or [workspace.metadata].
```

Comment:
```suggestion
/// A `[cargo-semver-checks]` config table in `[package.metadata]` 
/// or `[workspace.metadata]`.
```
Let's quote items in doc comments that are in square brackets. Otherwise rustdoc may try to introduce intra-doc links there, or readers may think they are intended to be such links (and are just broken).